tests/mpi/tria_signals_01.cc

```diff
@@ -0,0 +1,254 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// test signals of Triangulation class in parallel
+
+#include "../tests.h"
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/distributed/tria.h>
+
+int signal_counter_create = 0;
+int signal_counter_pre_refinement = 0;
+int signal_counter_post_refinement = 0;
+int signal_counter_pre_coarsening_on_cell = 0;
+int signal_counter_post_refinement_on_cell = 0;
+int signal_counter_copy = 0;
+int signal_counter_clear = 0;
+int signal_counter_any_change = 0;
+
+std::ofstream logfile;
+
+void f_create ()
+{
+  ++signal_counter_create;
+  return;
+}
+
+void f_pre_refinement ()
+{
+  ++signal_counter_pre_refinement;
+  return;
+}
+
+void f_post_refinement ()
+{
+  ++signal_counter_post_refinement;
+  return;
+}
+
+template<int dim, int spacedim>
+void f_pre_coarsening_on_cell (const typename Triangulation<dim, spacedim>::cell_iterator &)
+{
+  ++signal_counter_pre_coarsening_on_cell;
+  return;
+}
+
+template<int dim, int spacedim>
+void f_post_refinement_on_cell (const typename Triangulation<dim, spacedim>::cell_iterator &)
+{
+  ++signal_counter_post_refinement_on_cell;
+  return;
+}
+
+template<int dim, int spacedim>
+void f_copy (const Triangulation<dim, spacedim> &)
+{
+  ++signal_counter_copy;
+  return;
+}
+
+void f_clear ()
+{
+  ++signal_counter_clear;
+  return;
+}
+
+void f_any_change ()
+{
+  ++signal_counter_any_change;
+  return;
+}
+
+void serial_output(std::ostringstream &log_string)
```

Comment:
This is much easier if you use ```MPILogInitAll log;``` in ```main()``` and just write your output into ```deallog```. Please see ```mpi/repartition_01.cc``` for an example.